### ISSUE TITLE
Js instantiates a c++virtual object method. How to call the js object method in the c++thread? If it is directly in the main, there is no problem, but the instance cannot be called in the thread

### ISSUE THREAD

**Issue by Pmeak-git** [emscripten-core/emscripten#20979]
This is pseudocode, and the fundamental issue is that I want to call the method of the object instance inherited from JavaScript in thread. This is possible in the main thread, but it seems that it cannot be run in the child thread. Any help would be greatly appreciated

**c++ code :**
`#include <iostream>
using namespace std;

void callback();

class CommonCallback {
public:
	static CommonCallback* call = nullptr;
	static void setCommonCallback(CommonCallback* call){
		this.call = call;
	}
	static void callback(){
		auto t = new std::thtread(&callback);
		t.detach();
	}
	virtual void onComplete(int32_t code, const String &message){};
}

void callback(){
	CommonCallback::call.onComplete(1,"ddddddd");
}

int main(){
   cout << "Hello World";
   return 0;
}`

**javascript:**
`var CommonCallbackIMPL = Module.CommonCallback.extend("CommonCallback",{
    onComplete: function (code, message) {
        console.log('登录'+(code===0?"成功":"失败,messsage: " + message))
    }
});
`
**Browser error:**
![image](https://github.com/emscripten-core/emscripten/assets/69860588/e9bf7cfc-182e-4231-8437-528edfc70651)
